iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -124,38 +122,13 @@ private async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri,
                 if (Logging.IsEnabled)
                     Logging.Enter(this, nameof(CreateClientWebSocketAsync));
 
-                var websocket = new ClientWebSocket();
+                bool customWebSocket = _amqpTransportSettings.ClientWebSocket != null;
+                ClientWebSocket websocket = customWebSocket ? _amqpTransportSettings.ClientWebSocket : new ClientWebSocket();
+                IWebProxy webProxy = customWebSocket ? websocket.Options.Proxy : _amqpTransportSettings.Proxy;
 
                 // Set SubProtocol to AMQPWSB10
                 websocket.Options.AddSubProtocol(WebSocketConstants.SubProtocols.Amqpwsb10);
 
-                // Check if we're configured to use a proxy server
-                IWebProxy webProxy = _amqpTransportSettings.Proxy;
-
-                try
-                {
-                    if (webProxy != DefaultWebProxySettings.Instance)
-                    {
-                        // Configure proxy server
-                        websocket.Options.Proxy = webProxy;
-                        if (Logging.IsEnabled)
-                            Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Set ClientWebSocket.Options.Proxy to {webProxy}");
-                    }
-                }
-                catch (PlatformNotSupportedException)
-                {
-                    // .NET Core 2.0 doesn't support proxy. Ignore this setting.
-                    if (Logging.IsEnabled)
-                        Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as .NET Core 2.0 doesn't support proxy");
-                }
-
-                if (_amqpTransportSettings.WebSocketKeepAlive.HasValue)
```

Comment:
Is this no longer being set on the webSocket? The customer likely won't know to set this, so we should check their supplied websocket instance to see if they have set it, else set it for them. In the case where we create the web socket for them, we still need to set it.